jenkins_pipelines/environments/common/pipeline-pull-request.groovy

```diff
@@ -228,38 +232,46 @@ def run(params) {
                 echo "DEBUG: Deploy 1"
                 ws(environment_workspace){
                     if(must_test) {
+                        // Delete old terraform.tfvars
+                        sh "rm -f ${env.resultdir}/sumaform/terraform.tfvars"
+                        // Merge product en platform variables into terraform.tfvars
+                        sh "cat ${tfvars_product_version} ${tfvars_platform_localisation} >> ${env.resultdir}/sumaform/terraform.tfvars"
+                        // Add environment to use in tfvars
+                        sh "echo 'ENVIRONMENT = \'${env_number}\'' >> ${env.resultdir}/sumaform/terraform.tfvars"
+                        // Copy the variable declaration file
+                        sh "cp ${tf_local_variables} ${env.resultdir}/sumaform/"
+
+                        // Add all repositories variables
                         // Passing the built repository by parameter using a environment variable to terraform file
                         // TODO: We will need to add a logic to replace the host, when we use IBS for spacewalk
-                        env.PULL_REQUEST_REPO= "http://${fqdn_jenkins_node}/workspace/${short_product_name}-pr${env_number}/repos/${builder_project}:${pull_request_number}/${build_repo}/${arch}"
-                        env.MASTER_REPO = "http://${fqdn_jenkins_node}/workspace/${short_product_name}-pr${env_number}/repos/${source_project}/${build_repo}/${arch}"
-                        env.MASTER_OTHER_REPO = "http://${fqdn_jenkins_node}/workspace/${short_product_name}-pr${env_number}/repos/${other_project}/${other_build_repo}/${arch}"
-                        env.MASTER_SUMAFORM_TOOLS_REPO = "http://${fqdn_jenkins_node}/workspace/${short_product_name}-pr${env_number}/repos/${sumaform_tools_project}/${build_repo}/${arch}"
-                        env.TEST_PACKAGES_REPO = "http://${fqdn_jenkins_node}/workspace/${short_product_name}-pr${env_number}/repos/${test_packages_project}/rpm/${arch}"
-                        env.UPDATE_REPO = update_repo
+                        sh "echo \"############ Repositories variables ############\" >> ${env.resultdir}/sumaform/terraform.tfvars"
+                        sh "echo PULL_REQUEST_REPO = \\\"http://${fqdn_jenkins_node}/workspace/${short_product_name}-pr${env_number}/repos/${builder_project}:${pull_request_number}/${build_repo}/${arch}\\\" >> ${env.resultdir}/sumaform/terraform.tfvars"
```

Comment:
I did not check, but I would say we need to escape other characters inside that URL 🤔

Reply:
Not it works correctly :
```
PULL_REQUEST_REPO = "http://jenkins-worker-prs.mgr.prv.suse.net/workspace/suma-pr10/repos/systemsmanagement:Uyuni:Master:PR:8067/openSUSE_Leap_15.5/x86_64"
MASTER_REPO = "http://jenkins-worker-prs.mgr.prv.suse.net/workspace/suma-pr10/repos/systemsmanagement:Uyuni:Master/openSUSE_Leap_15.5/x86_64"
MASTER_OTHER_REPO = "http://jenkins-worker-prs.mgr.prv.suse.net/workspace/suma-pr10/repos/systemsmanagement:Uyuni:Master:Other/openSUSE_Leap_15.5/x86_64"
MASTER_SUMAFORM_TOOLS_REPO = "http://jenkins-worker-prs.mgr.prv.suse.net/workspace/suma-pr10/repos/systemsmanagement:sumaform:tools/openSUSE_Leap_15.5/x86_64"
TEST_PACKAGES_REPO = "http://jenkins-worker-prs.mgr.prv.suse.net/workspace/suma-pr10/repos/systemsmanagement:Uyuni:Test-Packages:Pool/rpm/x86_64"
UPDATE_REPO = "http://minima-mirror-ci-bv.mgr.prv.suse.net/jordi/some-updates/"
ADDITIONAL_REPO_URL = "http://minima-mirror-ci-bv.mgr.prv.suse.net/jordi/dummy/"
SLE_CLIENT_REPO = "http://jenkins-worker-prs.mgr.prv.suse.net/workspace/suma-pr10/repos/systemsmanagement:Uyuni:Master:SLE15-Uyuni-Client-Tools/SLE_15/x86_64"
RHLIKE_CLIENT_REPO = "http://jenkins-worker-prs.mgr.prv.suse.net/workspace/suma-pr10/repos/systemsmanagement:Uyuni:Master:EL9-Uyuni-Client-Tools/EL_9/x86_64"
DEBLIKE_CLIENT_REPO = "http://jenkins-worker-prs.mgr.prv.suse.net/workspace/suma-pr10/repos/systemsmanagement:Uyuni:Master:Ubuntu2204-Uyuni-Client-Tools/xUbuntu_22.04/x86_64"
OPENSUSE_CLIENT_REPO = "http://jenkins-worker-prs.mgr.prv.suse.net/workspace/suma-pr10/repos/systemsmanagement:Uyuni:Master:openSUSE_Leap_15-Uyuni-Client-Tools/openSUSE_Leap_15.0/x86_64"
```